datafusion/physical-plan/src/aggregates/group_values/metrics.rs

```diff
@@ -0,0 +1,204 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Metrics for the various group-by implementations.
+
+use crate::metrics::{ExecutionPlanMetricsSet, MetricBuilder, Time};
+
+pub struct GroupByMetrics {
+    pub aggregate_arguments_time: Time,
+    pub aggregation_time: Time,
+    pub emitting_time: Time,
```

Comment:
Because you don't expose the metrics outside the crate you should specify `pub(crate)` to avoid confusion and accidental leakage of information

```suggestion
pub(crate) struct GroupByMetrics {
    pub(crate) aggregate_arguments_time: Time,
    pub(crate) aggregation_time: Time,
    pub(crate) emitting_time: Time,
```